client/app/prouter.go

```diff
@@ -29,6 +35,11 @@ func makeProcessProposalRouter(app *App) *baseapp.MsgServiceRouter {
 // It also updates some external state.
 func makeProcessProposalHandler(router *baseapp.MsgServiceRouter, txConfig client.TxConfig) sdk.ProcessProposalHandler {
 	return func(ctx sdk.Context, req *abci.RequestProcessProposal) (*abci.ResponseProcessProposal, error) {
+		// Only allow 10s to process a proposal. Reject proposal otherwise.
+		timeoutCtx, timeoutCancel := context.WithTimeout(ctx.Context(), processTimeout)
```

Comment:
Here in the comment it says allowing 10s to process, but processTimeout is set to 1 Minute. Do I miss anything here?